packages/renderer/src/lib/statusbar/ProviderWidget.svelte

```diff
@@ -0,0 +1,73 @@
+<script lang="ts">
+import { router } from 'tinro';
+
+import type { ProviderInfo } from '/@api/provider-info';
+
+import IconImage from '../appearance/IconImage.svelte';
+
+interface Props {
+  entry: ProviderInfo;
+  command?: () => void;
+}
+
+let { entry, command = () => router.goto('/preferences/resources') }: Props = $props();
+
+let statusStyle = $derived.by(() => {
+  if (entry.containerConnections.length > 0) {
+    return (
+      statusesStyle.get(entry.containerConnections[0].status) ??
+      'bg-[var(--pd-status-not-running)] text-[var(--pd-status-not-running)]'
+    );
+  } else if (entry.kubernetesConnections.length > 0) {
+    return (
+      statusesStyle.get(entry.kubernetesConnections[0].status) ??
+      'bg-[var(--pd-status-not-running)] text-[var(--pd-status-not-running)]'
+    );
+  } else {
+    return statusesStyle.get(entry.status) ?? 'bg-gray-900 text-[var(--pd-content-text)]';
+  }
+});
+
+function tooltipText(entry: ProviderInfo): string {
+  let tooltip = '';
+  if (entry.containerConnections.length > 0) {
+    tooltip = entry.containerConnections.map(c => c.name).join(', ');
+  } else if (entry.kubernetesConnections.length > 0) {
+    tooltip = entry.kubernetesConnections.map(c => c.name).join(', ');
+  }
+  return tooltip;
+}
```

Comment:
question: why do we have a function there and we call then tooltipText(entry) at the bottom ?

issue: It looks like we do have a single entry for the whole object, so why do we need to give it as a parameter ?

it shoul be simplified by

`let tooltipText = $derived(....)`

and later `title={tooltipText}`